pkg/antctl/raw/check/installation/test_egressdenyall.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2024 Antrea Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package installation
+
+import (
+	"context"
+	"fmt"
+
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type EgressDenyAllConnectivityTest struct{}
+
+func init() {
+	RegisterTest("egress-deny-all-connectivity", &EgressDenyAllConnectivityTest{})
+}
+
+func (a EgressDenyAllConnectivityTest) Run(ctx context.Context, testContext *testContext) error {
+	services := []string{echoSameNodeDeploymentName}
+	if testContext.echoOtherNodePod != nil {
+		services = append(services, echoOtherNodeDeploymentName)
+	}
+	if err := applyEgressDenyAll(ctx, testContext.client, testContext.namespace); err != nil {
+		return err
+	}
+	testContext.Log("NetworkPolicy applied successfully")
+	for _, clientPod := range testContext.clientPods {
+		for _, service := range services {
+			if err := testContext.runAgnhostConnect(ctx, clientPod.Name, "", service, 80); err != nil {
+				testContext.Log("NetworkPolicy is working as expected: Pod %s cannot connect to Service %s", clientPod.Name, service)
+			} else {
+				return fmt.Errorf("networkPolicy is not working as expected: Pod %s connected to Service %s when it should not", clientPod.Name, service)
+			}
+		}
+	}
+	if err := testContext.client.NetworkingV1().NetworkPolicies(testContext.namespace).Delete(ctx, "egress-deny-all", metav1.DeleteOptions{}); err != nil {
```

Comment:
This should be called via `defer` right after `applyEgressDenyAll` succeeds, otherwise it wouldn't be deleted if the test fails, affecting following tests.